plugins/modules/digital_ocean_domain_record_info.py

```diff
@@ -0,0 +1,227 @@
+  community.digitalocean.digital_ocean_domain_record_info:
+    state: present
+    oauth_token: xxxx
+    domain: example.com
+
+- name: Get specific domain record by ID
+  community.digitalocean.digital_ocean_domain_record_info:
+    state: present
+    oauth_token: xxxx
+    record_id: 12345789
+  register: result
+
+- name: Retrieve all A domain records for example.com
+  community.digitalocean.digital_ocean_domain_record_info:
+    state: present
+    oauth_token: xxxx
+    domain: example.com
+    type: A
+"""
+
+RETURN = r"""
+data:
+  description: list of DigitalOcean domain records
+  returned: success
+  type: list
+  elements: dict
+  sample:
+    - data: ns1.digitalocean.com
+      flags: null
+      id: 296972269
+      name: '@'
+      port: null
+      priority: null
+      tag: null
+      ttl: 1800
+      type: NS
+      weight: null
+"""
+
+
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.digitalocean.plugins.module_utils.digital_ocean import (
+    DigitalOceanHelper,
+)
+
+
+class DigitalOceanDomainRecordManager(DigitalOceanHelper, object):
+    def __init__(self, module):
+        super(DigitalOceanDomainRecordManager, self).__init__(module)
+        self.module = module
+        self.domain = module.params.get("name").lower()
+        self.records = self.__get_all_records()
+        self.payload = self.__build_payload()
+        self.force_update = module.params.get("force_update", False)
+        self.record_id = module.params.get("record_id", None)
+        self.records_by_id = self.__find_record_by_id(self.record_id)
+
+    def check_credentials(self):
+        # Check if oauth_token is valid or not
+        response = self.get("account")
+        if response.status_code == 401:
+            self.module.fail_json(
+                msg="Failed to login using oauth_token, please verify validity of oauth_token"
+            )
+
+    def __get_all_records(self):
+
+        records = []
+        page = 1
+        while True:
+            # GET /v2/domains/$DOMAIN_NAME/records
+            type = self.module.params.get("type")
+            if type:
+                response = self.get(
+                    "domains/%(domain)s/records?type=%(type)s&page=%(page)s"
+                    % {"domain": self.domain, "type": type, "page": page}
+                )
+            else:
+                response = self.get(
+                    "domains/%(domain)s/records?page=%(page)s"
+                    % {"domain": self.domain, "page": page}
+                )
+            status_code = response.status_code
+            json = response.json
+
+            if status_code != 200:
+                self.module.exit_json(
+                    msg="Error getting domain records [%(status_code)s: %(json)s]"
+                    % {"status_code": status_code, "json": json}
+                )
+
+            for record in json["domain_records"]:
```

Comment:
Potential `KeyError` here if `domain_records` does not exist in `json`. (Maybe prefer `.get()`.)

Reply:
I've pushed changes.